12-watchtower-API.md

```diff
@@ -44,41 +46,37 @@ Connections between the client and the server can be long-lived or restarted for
 
 ### The `wt_init` message
 
-This messsage contains the information about a node and the type of appointments he is willing to create / accept.
+This message contains the information about a node and the type of appointments he is willing to create / accept.
 
-1. type: 128 (`wt_init`)
+1. type: ? (`wt_init`)
 2. data:
-   * [`u16`:`client_server`]
    * [`u16`:`aclen`]
    * [`aclen*bytes`:`accepted_ciphers`]
    * [`u16`:`modlen`]
    * [`modlen*bytes`:`modes`]
    * [`u16`:`qoslen`]
 	* [`qoslen*bytes`:`qos`]	
 
-`client_server` defines whether the sender is a client or a server.
-
-`accepted_ciphers` define the ciphers that the sender accepts and that he can use to encrypt / decrypt data. Accepted cyphers include `chacha20` and `aes-gcm-256`.
+`accepted_ciphers` define the ciphers that the sender implements and that he can use to encrypt / decrypt data. Accepted cyphers include `chacha20` and `aes-gcm-256`.
 
-`modes` define the operation mode requested / accepted. Modes include `altruistic` and `non-altuistic`.
+`modes` define the operation mode requested / accepted. Modes include `altruistic` and `non-altruistic`.
 
 `qos` defines whether the sender is requesting / accepting Quality of Service for his appointments. The only QoS offered at the moment is `accountability`.
```

Comment:
Is you refer to the QoS, this is optional, so lack of it is None.

Reply:
yeah i think it should be mandatory

Reply:
Why? If there is no QoS sending a message with None is redundant. If a service is optional it makes more sense than the field is also optional.

Reply:
just makes deserialisation easier? i.e. no need to worry about optional parameters? what do they do in the current standards for that?

Reply:
It doesn't actually. A parser goes: param1_len:param:param2_len:param:...

So it will stop when it does not find anything else. Adding a field with None just wastes bandwidth and makes the parser have to work more for every single message.

Anyway, if we end up doing the watchtower discovery instead of two wt_init (which seems way better) this won't be a problem since the tower will announce their services straightaway.

Reply:
ok make changes and update this issue with resolution